app/lib/actions/candlepin/listen_on_candlepin_events.rb

```diff
@@ -125,12 +136,17 @@ def connect_listening_service(event)
 
       def poll_listening_service(_event)
         output[:connection] = "Connected"
+        output[:error] = nil
         suspend do |suspended_action|
           CandlepinListeningService.instance.poll_for_messages(SuspendedAction.new(suspended_action))
         end
       end
 
       def initialize_service
+        at_exit do
```

Comment:
The `before_termination` hooks should be preferred to the at_exit, as the dynflow itself is handling the at_exit to terminate itself. see https://github.com/jlsherrill/katello/blob/11166/app/lib/actions/candlepin/listen_on_candlepin_events.rb#L162 (new in dynflow 0.8), also, since we register this in https://github.com/jlsherrill/katello/blob/11166/app/lib/actions/candlepin/listen_on_candlepin_events.rb#L162, is this code even needed?